assets/javascripts/job_templates.js

```diff
@@ -106,7 +106,14 @@ function addTestRow() {
     $('<td class="prio">50</td>').appendTo(tr);
 
     var archnames = table.data('archs');
-    $.each(archnames, function(ai, arch) {
+    var archHeaders = table.find('thead th.arch');
+    var archColIndex = 0;
+    $.each(archnames, function(archIndex, arch) {
+        while (archColIndex < archHeaders.length
```

Comment:
This while loop is the fix. The other changed are mostly refactoring because I thought it would be required in the first place.